### PR TITLE
Improving README and configuration instructions

### PR DESCRIPTION
**Pull request by darrensapalo**
These were the additional steps that I found was necessary to configure the application locally on my Macbook M2 machine.

## Status
Ready for review

## Change request overview

**These include:**
- Missing dependencies on Dockerfile
- Database and table definitions on Supabase that was missing included in README
- Revisions to include more precise instructions on README

## Recommendations

My commit log is messy; highly recommend to squash my changes before merging to `main`.

## Developer notes
I'm not really a python developer. For Python programming ability, I would qualify myself between [stage 3 and stage 4](https://protocol.tome.gg/training/definition) skill level. 

I stumbled upon this project because my interests are in EdTech. This project is really promising. As I don't use Python for production level apps, please give feedback on coding best practices on Python.

### CODE REVIEW

**Comment by darrensapalo** [comment]
Successfully removed irrelevant changes (to be added in a separate PR).

**Things that this pull request ought not to include:**

- Partial work on adding unit tests
- Abstracting the mediator into an abstract base class, and having the supabase mediator implement it
- Some minor prompting tweaks intended for use in Tome.gg
- Partial work on adding a new mediator (not supabase, but a generalized postgre connector)

File: Dockerfile [hunk 1]
```diff
@@ -14,6 +14,10 @@ ENV PYTHONFAULTHANDLER=1 \
   POETRY_VERSION=1.4.1
 
 RUN pip install "poetry==$POETRY_VERSION"
+RUN pip install streamlit
```
Comment (VVoruganti) [inline Dockerfile:17]: The project is no longer using streamlit for the web-ui so there shouldn't be a need to add streamlit

File: agent/supabasemediator.py
Comment (VVoruganti) [inline agent/supabasemediator.py:1]: As you've highlighted above a user doesn't necessarily need to be using supabase for memory of the application. The mediator file was planned to contain multiple classes for different storage solutions mongo, supabase, postgres.

To that end the preference would be to keep the name as mediator. There is potential to instead create a sub modules with multiple different files and an `__init__.py` to support the different mediators.

File: agent/cache.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 in OrderedDict data structure.
 """
 from collections import OrderedDict
-from .mediator import SupabaseMediator
+from .supbase_mediator import SupabaseMediator
```
Comment (VVoruganti) [inline agent/cache.py:6]: See below comment

**Comment by VVoruganti** [comment]
Can you also change this PR to target the `staging` branch rather than main. Changes need to go into main before we put them into the upstream on `main`

**Comment by vintrocode** [comment]
Does #121 resolve this? @VVoruganti

**Comment by VVoruganti** [comment]
Closing this as stale

**Pull request closed without merging**